src/main/scala/com/crealytics/spark/excel/ExcelRelation.scala

```diff
@@ -145,13 +160,16 @@ case class ExcelRelation(
   }
 
   private def inferSchema(): StructType = this.userSchema.getOrElse {
-    val headerIndices = headerCells.map(_.getColumnIndex)
-    val rawHeader = headerCells.map(_.getStringCellValue)
+    val rawHeader = headerCells.map(getCellValue)
 
     val dataTypes = if (this.inferSheetSchema) {
       val stringsAndCellTypes: Seq[Seq[DataType]] = excerpt.tail
         .map { r =>
-          headerIndices.map(i => r.find(_.getColumnIndex == i).map(getSparkType).getOrElse(DataTypes.NullType))
+          headerCells.map(
```

Comment:
Just to understand this: Should the removal of `headerIndices` change the logic in some way, or is this to reduce the number of local variables?

Reply:
There was 2 iterations before. 1st iteration to get indices and 2nd iteration to get sparkType.
Now it is 1 iteration.